doc/developers/developing_scikit.rst

```diff
@@ -0,0 +1,727 @@
+the corresponding subpackages.
+
+We expect code coverage of new features to be at least around 90%.
+
+For guidelines on how to use ``pytest`` efficiently, see the
+:ref:`pytest_tips`.
+
+Writing matplotlib related tests
+--------------------------------
+
+Test fixtures ensure that a set of tests will be executing with the appropriate
+initialization and cleanup. The scikit-learn test suite implements a fixture
+which can be used with ``matplotlib``.
+
+``pyplot``
+    The ``pyplot`` fixture should be used when a test function is dealing with
+    ``matplotlib``. ``matplotlib`` is a soft dependency and is not required.
+    This fixture is in charge of skipping the tests if ``matplotlib`` is not
+    installed. In addition, figures created during the tests will be
+    automatically closed once the test function has been executed.
+
+To use this fixture in a test function, one needs to pass it as an
+argument::
+
+    def test_requiring_mpl_fixture(pyplot):
+        # you can now safely use matplotlib
+
+Workflow to improve test coverage
+---------------------------------
+
+To test code coverage, you need to install the `coverage
+<https://pypi.org/project/coverage/>`_ package in addition to pytest.
+
+1. Run 'make test-coverage'. The output lists for each file the line
+    numbers that are not tested.
+
+2. Find a low hanging fruit, looking at which lines are not tested,
+    write or adapt a test specifically for these lines.
+
+3. Loop.
+
+.. _coding-guidelines:
+
+Coding guidelines
+=================
+
+The following  are some standards used within scikit-learn, which may
+be appropriate to adopt in external projects. Of course, there are special
+cases and there will be exceptions to these rules. However, following
+these rules when submitting new code makes the review easier so new code
+can be integrated in less time.
+
+Uniformly formatted code makes it easier to share code ownership. The
+scikit-learn project tries to closely follow the official Python guidelines
+detailed in `PEP8 <https://www.python.org/dev/peps/pep-0008>`_ that
+detail how code should be formatted and indented. Please read it and
+follow it.
+
+In addition, we add the following guidelines:
+
+* Use underscores to separate words in non class names: ``n_samples``
+  rather than ``nsamples``.
+
+* Avoid multiple statements on one line. Prefer a line return after
+  a control flow statement (``if``/``for``).
+
+* Use relative imports for references inside scikit-learn.
+
+* Unit tests are an exception to the previous rule;
+  they should use absolute imports, exactly as client code would.
+  A corollary is that, if ``sklearn.foo`` exports a class or function
+  that is implemented in ``sklearn.foo.bar.baz``,
+  the test should import it from ``sklearn.foo``.
+
+* **Please don't use** ``import *`` **in any case**. It is considered harmful
+  by the `official Python recommendations
+  <https://docs.python.org/2/howto/doanddont.html#from-module-import>`_.
```

Comment:
This URL should be updated: https://docs.python.org/3.1/howto/doanddont.html#at-module-level

Reply:
@jnothman Thanks for the feedback. I made all the changes you recommended above.